owncloudApp/src/main/java/com/owncloud/android/ui/preview/PreviewTextFragment.kt

```diff
@@ -0,0 +1,451 @@
+import java.io.BufferedWriter
+import java.io.FileInputStream
+import java.io.IOException
+import java.io.StringWriter
+import java.util.LinkedList
+import java.util.Scanner
+
+class PreviewTextFragment : FileFragment() {
+    private var mAccount: Account? = null
+    private lateinit var mProgressBar: ProgressBar
+    private lateinit var mProgressController: TransferProgressController
+    private lateinit var mTextPreview: TextView
+    private lateinit var mTextLayout: View
+    private lateinit var mTabLayout: TabLayout
+    private lateinit var mViewPager2: ViewPager2
+    private lateinit var rootView: RelativeLayout
+    private lateinit var mTextLoadTask: TextLoadAsyncTask
+
+    private val previewTextViewModel by viewModel<PreviewTextViewModel>()
+    private val fileOperationsViewModel by viewModel<FileOperationsViewModel>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+        Timber.v("onCreateView")
+
+        val ret = inflater.inflate(R.layout.preview_text_fragment, container, false)
+        ret.filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(context)
+
+        // We can use binding instead of findViewById
+        rootView = ret.findViewById(R.id.top)
+        mProgressBar = ret.findViewById(R.id.syncProgressBar)
+        mTabLayout = ret.findViewById(R.id.tab_layout)
+        mViewPager2 = ret.findViewById(R.id.view_pager)
+        mTextPreview = ret.findViewById(R.id.text_preview)
+        mTextLayout = ret.findViewById(R.id.text_layout)
+
+        return ret
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val file: OCFile?
+        if (savedInstanceState == null) {
+            val args = requireArguments()
+            file = args.getParcelable(EXTRA_FILE)
+            mAccount = args.getParcelable(EXTRA_ACCOUNT)
+            if (file == null) {
+                throw IllegalStateException("Instanced with a NULL OCFile")
+            }
+
+            if (mAccount == null) {
+                throw IllegalStateException("Instanced with a NULL ownCloud Account")
+            }
+        } else {
+            file = savedInstanceState.getParcelable(EXTRA_FILE)
+            mAccount = savedInstanceState.getParcelable(EXTRA_ACCOUNT)
+        }
+        setFile(file)
+        setHasOptionsMenu(true)
+        isOpen = true
+        currentFilePreviewing = file
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        mProgressController = TransferProgressController(mContainerActivity)
+        mProgressController.setProgressBar(mProgressBar)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        loadAndShowTextPreview()
+    }
```

Comment:
In this case it's no necessary because it's an empty method. In addition, I removed `super.onSaveInstanceState (outState)`, `onCreateOptionsMenu (menu)`, `onPrepareOptionsMenu (menu)` because they are also empty methods